jsonrpc/src/methods/state.json

```diff
@@ -139,6 +139,26 @@
       "$ref": "#/components/contentDescriptors/RecursiveFindContentResult"
     }
   },
+  {
```

Comment:
Do I understand correctly that these calls walk down the state to get to a specific account / storage data?
So basically a bunch of different `recursiveFindContent` calls?

Reply:
I would say that benefits are slightly better:

- the Execution api-s actually don't allow block hash. They explicitly support block number or string ("latest", "earliest" or "pending"), and I don't think we support either (I assume we plan to in the long run, so this might not be so big issue)
- official api for the `eth_getStorageAt` would have to always iterate over state trie (to find account state) and contract storage trie. The api that I propose splits those tree traversals into separate calls (meaning, if you want to get multiple values from the same contract, you will iterate state trie only once).

I'm fine with creating different namespace and / or making them optional. Maybe we can discuss this on the next team meeting.

Reply:
Further thinking about it, having a different namespace is not just useful for us in keeping some logic separate, but even more interesting for end users. 

Providing just one API which contains also all these lower level calls that are only useful for testing the actual protocol will be overwhelming and not great UX. Separate user focused API does sound better for that.